### DISCUSSION TITLE
Configuratuon Migration tool based on ServiceClient

### DISCUSSION THREAD

**Discussion by anandmyadam** [microsoft/PowerPlatform-DataverseServiceClient#267] [Q&A] [answered]
Any plans on updating the Configuration Migration tool to use this ServiceClient object? 

We have a console app created using this latest ServiceClient object and we want build the DataImport logic using the Configuration Migration dependant dlls.

**Comment by MattB-msft** [comment, marked as answer]
Disclaimer: We do not support ( or document ) how to use the configuration migration assemblies directly.   As such we do not support any such use of that tool outside of our supported clients. 

To your question, 
Short term, no.  We do not plan to switch it over to the ServiceClient.  CrmServiceClient smoothy supports both onPrem and OnLine Scenarios, were Dataverse ServiceClient is more suited to online scenarios, Primary due to authentication issues between MSAL and ADFS. 
That said we will likely support it via the CLI at some point in the future.

**Reply by anandmyadam** [reply]
Thank you @MattB-msft .
